legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -3438,10 +3451,30 @@ function meta::relational::functions::pureToSqlQuery::relationalmappingspecifica
    $vs->match([
                r:RootRelationalInstanceSetImplementation[1] | $r->dataTypePropertyMappings()
                                                               ->filter(x | if($state.graphFetchFlow == true, | $x.property->in($state.graphFetchProperties), | true))
-                                                              ->map(pm|pair($pm.property.name->toOne(),$pm->cast(@RelationalPropertyMapping).relationalOperationElement)),
+                                                              ->map(pm|   pair($pm.property.name->toOne(), buildPossibleEnumMappingPushDown($pm->cast(@RelationalPropertyMapping),$state).relationalOperationElement);)
+                                                                          , 
                v:View[1] | $v.columnMappings->map(cm|pair($cm.columnName, $cm.relationalOperationElement));
               ]);
 }
+function meta::relational::functions::pureToSqlQuery::buildPossibleEnumMappingPushDown(mapping: RelationalPropertyMapping[1],  state: State[1]): RelationalPropertyMapping[1]
+{ 
+  let relationalElement = $mapping.relationalOperationElement;
+   if($state.pushDownEnumTransformations && $mapping.transformer ->isNotEmpty() &&   $mapping.transformer->toOne()->instanceOf(EnumerationMapping)  && !$state.inFilter ,  
+       |let caseParams = $mapping.transformer->cast(@EnumerationMapping<Any>).enumValueMappings
+                                                      ->map(ev  |   if($ev.sourceValues->size() == 1,
+                                                                      | ^DynaFunction(name = 'equal', parameters=[$relationalElement, ^Literal(value=$ev.sourceValues->at(0))]),
+                                                                      | ^DynaFunction(name = 'in', parameters = [$relationalElement, ^LiteralList(values=$ev.sourceValues->map(v|^Literal(value=$v)))]);
+                                                                    )->concatenate(^Literal(value=$ev.enum.name));
+                                                            );
+      
+        ^$mapping(transformer = [], relationalOperationElement=^DynaFunction(name = 'case', parameters =   $caseParams->concatenate(^DynaFunction(name = 'sqlNull'))));
```

Comment:
This returning null seems to deviate from the current behavior, where we want to fail if a value on the database does not map to any enum.  While I'm not sure how to fail on SQL, but if returning null will be the way fwd when pushing down enums, we might want to reevaluate our in-memory implementation so that we try to be consistent.

Reply:
This matches the current behavior of TDS queries